### PR TITLE
Update to diesel 2.0.0-rc.1

### PR DESCRIPTION
**Pull request by weiznich**
This currently requires an additional patch for overwriting the
`QueryFragment` impl for `SelectStatement`. I will work on submitting a
PR for upstreaming this soon.

This only updates the implementation, not the tests yet. That needs to be done by someone that is more familiar with the firebird database system.

### CODE REVIEW

File: rsfbclient-diesel/src/lib.rs
Comment (weiznich) [inline rsfbclient-diesel/src/lib.rs:3]: As diesel maintainer I would prefer not to reexport diesel here. Instead only the custom connection/backend implementation should be provided + users should just depend on diesel directly. That gives them much greater control about enabled features + makes it clearer where stuff is coming from.

**Comment by fernandobatels** [comment]
Thanks for your contribution. I was already studying the diesel rc to fix out implementation, your work will help a lot.

> That needs to be done by someone that is more familiar with the firebird database system.

Don’t worry, I got this.

**Comment by weiznich** [comment]
> Don’t worry, I got this.

That would be great. Just let me know if there is something unclear form diesels side.

File: rsfbclient-diesel/src/lib.rs
Comment (fernandobatels) [reply rsfbclient-diesel/src/lib.rs:3]: I will check this before finish the PR

**Comment by fernandobatels** [comment]
Todo:
- [x] fix tests connection
- [ ] fix tests crud 
- [ ] fix tests query
- [ ] fix tests types
- [ ] check the diesel re-export
- [ ] wait for diesel-rs/diesel#3257 and update the reference
- [ ] revert the commit 873b948667840ac27c549444f5180351d9ea1288

**Comment by fernandobatels** [comment]
> That would be great. Just let me know if there is something unclear form diesels side.

Can you tell me what I'm missing?
``` 
error[E0275]: overflow evaluating the requirement `_: Sized`
    --> rsfbclient-diesel/src/tests/crud.rs:21:10
     |
21   |         .execute(conn)
     |          ^^^^^^^
     |
     = help: consider increasing the recursion limit by adding a `#![recursion_limit = "256"]` attribute to your crate (`rsfbclient_diesel`)
``` 

I already changed the recursion_limit, but without effect. I'm currently checking the implementation of the others backends.

**Comment by fernandobatels** [comment]
> > That would be great. Just let me know if there is something unclear form diesels side.
> 
> Can you tell me what I'm missing?
> 
> ```
> error[E0275]: overflow evaluating the requirement `_: Sized`
>     --> rsfbclient-diesel/src/tests/crud.rs:21:10
>      |
> 21   |         .execute(conn)
>      |          ^^^^^^^
>      |
>      = help: consider increasing the recursion limit by adding a `#![recursion_limit = "256"]` attribute to your crate (`rsfbclient_diesel`)
> ```
> 
> I already changed the recursion_limit, but without effect. I'm currently checking the implementation of the others backends.

ToSql of i32 and i64 missing. Solved.

**Comment by weiznich** [comment]
> ToSql of i32 and i64 missing

Other `ToSql` impls are likely missing as well. Diesel does not provide any of these impls by default anymore.

**Comment by fernandobatels** [comment]
I will wait for diesel-rs/diesel#325, but this PR is ready for review

**Review by jairinhohw — APPROVED** [review]

**Pull request merged**
